app/server/models/credit_transfer.py

```diff
@@ -28,7 +24,7 @@ class CreditTransfer(ManyOrgBase, BlockchainTaskableBase):
     uuid            = db.Column(db.String, unique=True)
 
     resolved_date   = db.Column(db.DateTime)
-    transfer_amount = db.Column(db.Integer)
+    _transfer_amount = db.Column(db.BigInteger)
```

Comment:
transfer amounts on the blockchain have 256 bits of precision - needing to store some sub $0.01 amounts for exchanges now, so this gets us some of the way there. Problem is that even bigints cant handle the full range of the blockchain-native amounts, so we can't just store them across 1:1

Reply:
[pg-bignum](https://github.com/beargiles/pg-bignum) sounds like a valid option, alternatively, we could store in `numeric` and do a conversion in application code?

https://dba.stackexchange.com/questions/62934/adding-unsigned-256-bit-integers-in-postgresql/192746

Reply:
mmmmm yeah i did look in `numeric`, though I read somewhere that performance aint fantastic. I guess it would make it FAR easier to not need to store amounts at three levels of precision like we are now:
- int cents on the front end, effectively 2 decimals
- 256 bit int, effectively 18 decimals in contracts
- big int, effectively 6 + 2 = 8 decimals on the db
 🤢

Reply:
from https://www.postgresql.org/docs/10/datatype-numeric.html
on one hand...
> calculations on numeric values are very slow compared to the integer types, or to the floating-point types described in the next section.

on the other

>  It is especially recommended for storing monetary amounts

Reply:
thoughts @tristanhcole and @wangfowen? Seems to be a bit of debate about this on the web:
https://dba.stackexchange.com/questions/110880/numeric-vs-integer-for-a-column-size-and-performance

i'm inclined to go towards using numeric, because we've been stung by amounts not adding to 0 on contracts before, but about 49-51 split.

Reply:
beyond basic addition and subtraction, would there be extensive calculations? i haven't found any good benchmarks of exactly how much slower numeric is, but i can't imagine the added slowness causing that much of a difference. if storage of the number is less error prone i'm for it

Reply:
Ok let's do numeric!